### PR TITLE
[ISSUE #5568] Support long length group and topic for pop mode consumption

### PR DESCRIPTION
**Pull request by lizhimins**
see #5568

### CODE REVIEW

File: common/src/main/java/org/apache/rocketmq/common/message/MessageExtBrokerInner.java [hunk 1]
```diff
@@ -27,6 +27,8 @@ public class MessageExtBrokerInner extends MessageExt {
 
     private ByteBuffer encodedBuff;
 
+    private MessageVersion version = MessageVersion.MESSAGE_VERSION_V1;
+
```
Comment (fuyou001) [inline common/src/main/java/org/apache/rocketmq/common/message/MessageExtBrokerInner.java:31]: version default null  may be safe.

File: common/src/main/java/org/apache/rocketmq/common/message/MessageExtBrokerInner.java [hunk 2]
```diff
@@ -27,6 +27,8 @@ public class MessageExtBrokerInner extends MessageExt {
 
     private ByteBuffer encodedBuff;
 
+    private MessageVersion version = MessageVersion.MESSAGE_VERSION_V1;
```
Comment (fuyou001) [inline common/src/main/java/org/apache/rocketmq/common/message/MessageExtBrokerInner.java:30]: version default null may be safe.

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java [hunk 3]
```diff
@@ -586,9 +587,37 @@ private long popMsgFromQueue(boolean isRetry, GetMessageResult getMessageResult,
...
+                        try {
+                            String ckInfo = ExtraInfoUtil.buildExtraInfo(offset, popTime, requestHeader.getInvisibleTime(),
+                                reviveQid, messageExt.getTopic(), brokerName, messageExt.getQueueId(), messageExt.getQueueOffset());
+                            messageExt.getProperties().putIfAbsent(MessageConst.PROPERTY_POP_CK, ckInfo);
+
+                            // Set retry message topic to origin topic and clear message store size to recode
+                            messageExt.setTopic(requestHeader.getTopic());
+                            messageExt.setStoreSize(0);
+
+                            byte[] encode = MessageDecoder.encode(messageExt, true);
```
Comment (xdkxlk) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java:611]: message is not been decompressed at MessageDecoder.decodesBatch(mapedBuffer.getByteBuffer(),true, false, true)
and will be compressed here
which means, messages will be compressed twice

Comment (xdkxlk) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java:611]: In my opinion, this should be `MessageDecoder.encode(messageExt, false)`

**Review by xdkxlk — APPROVED** [review]

**Review by xdkxlk — APPROVED** [review]

**Pull request merged**